### PR TITLE
WV-2347 Updating Nightwatch Dependency from 1.6.2 to 2.3.3

### PR DESCRIPTION
**Pull request by ryanweiler92**
## Description

This updates our Nightwatch.js dependency from 1.6.2 to 2.3.3, the most current stable version. I had to update many of our end to end tests because of updates/changes to both Nightwatch.js and Selenium. Some of our tests were exhibiting some strange behaviors such as mouseButtonDown/mouseButtonUp/moveTo no longer working correctly. Most instances of these have been replaced with the latest user action creators that were released with the new version of Nightwatch. There are 2 tests that I had to make exceptions for on the Firefox side, layers-sidebar-test.js & timeline-mobile-test.js. I reached out to the creators of Nightwatch but they were unable to find the source of the errors for these tests. Hopefully in time there will be a solution to be able to reimplement both of these test in full on the Firefox side again. This also includes an update for the Gecko driver.

## How To Test

Install the latest packages and run the end to end tests.

### CODE REVIEW

File: e2e/features/compare/compare-mobile-test.js [hunk 1]
```diff
@@ -32,6 +32,7 @@ module.exports = {
 
   // load SWIPE and verify mobile compare A|B toggle buttons are visible
   'Mobile comparison A|B toggle buttons are visible and only A is selected by default': (c) => {
+    c.pause(10000);
```
Comment (jasontk19) [inline e2e/features/compare/compare-mobile-test.js:35]: This seems like a really long pause, is it necessary?

File: e2e/features/layers/layers-vector-test.js [hunk 2]
```diff
@@ -30,8 +30,8 @@ module.exports = {
     c.url(c.globals.url + damsLayerWMSZoomLevelQuerystring);
     c.waitForElementVisible('#active-GRanD_Dams .fa-hand-pointer', TIME_LIMIT, () => {
       c.moveToElement('#wv-map-geographic', 400, 200)
-        .mouseButtonClick(0);
-      c.pause(200);
+        .click('#wv-map-geographic');
+      c.pause(5000);
```
Comment (jasontk19) [inline e2e/features/layers/layers-vector-test.js:34]: This pause is also pretty long, can we not just `waitForElementVisible` on something here?

File: e2e/features/tour/tour-joyride-test.js [hunk 3]
```diff
@@ -19,17 +19,21 @@ module.exports = {
...
   },
   'Clicking beacon shows the floater tooltip': (c) => {
     if (c.options.desiredCapabilities.browserName !== 'firefox') { // intermittently fails on Chrome
       return;
     }
-    c.useCss().waitForElementVisible(firstBeaconSelector, TIME_LIMIT);
-    c.pause(500);
-    c.click(firstBeaconSelector);
+    // c.useCss().waitForElementVisible(firstBeaconSelector, TIME_LIMIT);
+    c.pause(5000);
```
Comment (jasontk19) [inline e2e/features/tour/tour-joyride-test.js:35]: Another one

**Review by jasontk19 — APPROVED** [review]
It would be good if we could remove the handful of really long pauses, otherwise I think this looks good and it's passing so... 👍🏻

File: e2e/features/compare/compare-mobile-test.js (hunk 1 above)
Comment (ryanweiler92) [reply e2e/features/compare/compare-mobile-test.js:35]: Removed pause, this was left over from debugging.

File: e2e/features/layers/layers-vector-test.js (hunk 2 above)
Comment (ryanweiler92) [reply e2e/features/layers/layers-vector-test.js:34]: Reduced pause, this was left over from debugging.

File: e2e/features/tour/tour-joyride-test.js (hunk 3 above)
Comment (ryanweiler92) [reply e2e/features/tour/tour-joyride-test.js:35]: Removed pause, this was left over from debugging.

**Pull request merged**